Update aarch64 makefiles

kubanrob opened the pull request:
This PR modifies the Makefiles to select the right sources for aarch64. It includes 3 commits from @marcrittinghaus.

This PR applies on top of PR #9 and should be used together with #6.

marcrittinghaus committed:
Remove files that depend on fork

Signed-off-by: Marc Rittinghaus <marc.rittinghaus@kit.edu>

marcrittinghaus committed:
Add translation from arm64 to aarch64

Signed-off-by: Marc Rittinghaus <marc.rittinghaus@kit.edu>

kubanrob commented:
`__set_thread_area.s` needs to be removed from makefile when merged into https://github.com/unikraft/lib-musl/pull/9. Will be removed.

kubanrob commented:
Update: Rebase to latest state of thread_support (PR #9)

razvanvirtan commented on Makefile.uk:
```diff
@@ -95,13 +97,13 @@ $(LIBMUSL)/src/internal/version.h:
+LIBMUSL_CLEAN += $(LIBMUSL)/arch/$(MUSL_ARCH)/bits/alltypes.h
 LIBMUSL_CLEAN += $(LIBMUSL)/src/internal/version.h
-LIBMUSL_CLEAN += $(LIBMUSL_BUILD)/include
+#LIBMUSL_CLEAN += $(LIBMUSL_BUILD)/include
```
Why do we remove this file? Is the reason connected to the goal of this commit?

razvanvirtan commented on Makefile.uk.musl.string:
```diff
@@ -91,6 +91,10 @@ LIBMUSL_STRING_SRCS-y += $(LIBMUSL)/src/string/arm/memcpy.c|arm
 LIBMUSL_STRING_SRCS-y += $(LIBMUSL)/src/string/memset.c|arm
+else ifeq (arm64,$(CONFIG_UK_ARCH))
+LIBMUSL_STRING_SRCS-y += $(LIBMUSL)/src/string/aarch64/memcpy.s|aarch64
+LIBMUSL_STRING_SRCS-y += $(LIBMUSL)/src/string/aarch64/memset.s|x86_64
```
Shouldn't this be `......|aarch64`?

kubanrob replied:
Yes, will update.

razvanvirtan commented on Makefile.uk:
```diff
@@ -95,13 +97,13 @@ $(LIBMUSL)/src/internal/version.h:
+LIBMUSL_CLEAN += $(LIBMUSL)/arch/$(MUSL_ARCH)/bits/alltypes.h
 LIBMUSL_CLEAN += $(LIBMUSL)/src/internal/version.h
-LIBMUSL_CLEAN += $(LIBMUSL_BUILD)/include
+#LIBMUSL_CLEAN += $(LIBMUSL_BUILD)/include
```
What is the purpose of removing this file? Is it connected to the goal of this commit?

kubanrob replied:
This is a directory, which breaks the clean rule. So probably should be removed properly, and maybe should live in a different PR.

kubanrob replied:
The new PR can be found here: PR #17

razvanvirtan commented:
Hi, Robert!
Besides the inline comments, there is a minor typo in https://github.com/unikraft/lib-musl/pull/7/commits/c3a54479bd51610b47032c35fff76ffc3fec0a53 (`configuration of`).

razvanvirtan reviewed (commented):
Hi, Robert!
Besides the inline comments, there is a minor typo in https://github.com/unikraft/lib-musl/pull/7/commits/c3a54479bd51610b47032c35fff76ffc3fec0a53 (`configuration of`).

eduardvintila reviewed (approved):
Hi, @kubanrob. Apart from what @razvanvirtan said, everything looks good to me.

Reviewed-by: Eduard Vintilă <eduard.vintila47@gmail.com>

kubanrob committed:
Add aarch64 configurations

This updates the sublibrary makefiles to use the appropiate
sourcefiles for aarch64.

It also removes the aarch64 source file for `__set_thread_area` from
the thread sublibrary makefile, as it is provided by lib-muslglue.

Signed-off-by: Robert Kuban <robert.kuban@opensynergy.com>

kubanrob commented:
**Update:** I removed the change to `LIBMUSL_CLEAN += $(LIBMUSL_BUILD)/include` and merged the two commits updating the sublibrary makefiles into one.

For the fixing the clean build target I will open a new PR, as commenting out / removing that line is not enough.

razvanvirtan commented:
Thank you, @kubanrob!
I also double checked the Makefile.uk.* files with your script and everything looks good.

Reviewed-by: Razvan Virtan <virtanrazvan@gmail.com>

razvand commented:
@razvanvirtan , please also approve the PR with the approve button.

razvanvirtan reviewed (approved):
Thank you, @kubanrob!
I also double checked the Makefile.uk.* files with your script and everything looks good.

Reviewed-by: Razvan Virtan <virtanrazvan@gmail.com>

razvand commented:
Thanks, @kubanrob . This works. I need to see whether I can approve it, or we should first integrated PR #9 and then you do a rebase.

kubanrob commented:
**Update**: rebased onto staging

razvand reviewed (approved):
Thanks, @kubanrob 

Reviewed-by: Razvan Deaconescu <razvand@unikraft.io>
Approved-by: Razvan Deaconescu <razvand@unikraft.io>
